anndata/tests/helpers.py

```diff
@@ -452,3 +491,29 @@ def fmt_name(x):
             exact,
             elem_name=fmt_name(attr),
         )
+
+
+# Dev. Note: Dask assumes behavior of numpy and sparse are identical.
+# Technically there shouldn't be a difference using this
+# or darr_from_arr_dense function, but there is in some cases.
+# See darr_from_arr_dense these cases.
+def darr_from_arr(arr):
+    import dask.array as da
+
+    return da.from_array(arr, chunks="auto")
+
+
+# Some tests use scipy.sparse matrices to create their
+# data. Unfortunately da.from_array doesn't work
+# properly if this is the case. To be sure that
+# dask from_array works as intended with the types
+# we use in AnnData this function can be helpful.
+# We wouldn't want all inputs to be called asarray
+# since we already assume they should have the same semantic.
+def darr_from_arr_dense(arr):
+    # For cases when we take scipy.sparse as input
+    # and don't care if it gets turned into a dense array.
+    # (e.g., comparing only the contents and not the type)
+    import dask.array as da
+
+    return da.from_array(asarray(arr), chunks="auto")
```

Comment:
```suggestion
@singledispatch
def as_dense_dask_array(a):
    import dask.array as da

    return da.asarray(a)


@as_dense_dask_array.register(sparse.spmatrix)
def _(a):
    return as_dense_dask_array(a.toarray())
```

And then a find and replace for all `darr_from_arr(_dense)?` to `as_dense_dask_array`.